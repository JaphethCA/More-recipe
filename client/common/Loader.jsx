import React from 'react';
import PropTypes from 'prop-types';
import { GridLoader } from 'react-spinners';

import './loader.scss';

/**
 * @description displays loading indicator
 * @param {Object} props
 * @returns {ReactElement} markup
 */
const Loader = props => (
  <div className="sweet-loading">
    <GridLoader
      color="rgb(223, 63, 63)"
      loading={props.isFetching}
    />
  </div>
);

Loader.propTypes = {
  isFetching: PropTypes.bool
};

Loader.defaultProps = {
  isFetching: false
};

export default Loader;
