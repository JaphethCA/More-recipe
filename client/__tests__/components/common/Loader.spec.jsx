import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';
import Loader from '../../../common/Loader';


const props = {
  isFetching: true
};

describe('<Loader /> ', () => {
  it('should render component without exploding', () => {
    const wrapper = shallow(<Loader {...props} />);
    expect(wrapper).toBeDefined();
    expect(wrapper.find('div').length).toBe(1);
  });

  it('should match snapshot', () => {
    const wrapper = shallow(<Loader {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

