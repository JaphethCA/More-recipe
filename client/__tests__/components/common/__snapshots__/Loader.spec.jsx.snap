// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader />  should match snapshot 1`] = `
<div
  className="sweet-loading"
>
  <onlyUpdateForKeys(Loader)
    color="rgb(223, 63, 63)"
    loading={true}
    margin="2px"
    size={15}
  />
</div>
`;
